test/route.test.js

```diff
@@ -46,3 +46,16 @@ test("Route without path prop matches to every location", () => {
   expect(Route({ render }, [])({ location: { pathname: "/" } })).toBe(true)
   expect(Route({ render }, [])({ location: { pathname: "/users" } })).toBe(true)
 })
+
+test("Route ignores url params containing invalid character sequences", () => {
+  const invalidChars = "%E0%A4%A"
+  expect(() => decodeURI(invalidChars)).toThrow(URIError)
+  const state = { location: { pathname: `/foo/${invalidChars}/bar/baz` } }
+  const actions = {}
+  const render = jest.fn(({ match }) => {
+    expect(match.params).toEqual({ bar: "baz" })
```

Comment:
@jorgebucaran Is this intended behavior? In my opinion, a malformed URI should not match to any routes. What do you think?

Reply:
Why not `toEqual({ foo: "%E0%A4%A", bar: "baz" })`

Reply:
Removing `decodeURI` from ↓ changes `match.params` to `{ foo: "%E0%A4%A", bar: "baz" }`. I'm not sure why `decodeURI` is used here.

https://github.com/hyperapp/router/blob/795eb9b81ad9a6027d74dc65b0fe9548b0470095/src/parseRoute.js#L30-L34

Reply:
Decoding is make sense because you usually expect `{ name: "café" }` instead of `{ name: "caf%C3%A9" }` in the params object.

but probably the code above should looks like this:
```js
function decodeParam(val) {
  try {
    return decodeURIComponent(val)
  } catch (err) {
    return val
  }
}

params[paths[i].slice(1)] = urls[i] = decodeParam(urls[i])
```

Reply:
> @yuku-t In my opinion, a malformed URI should not match to any routes. What do you think?

A lot of code was golfed to make the router as small as possible. It was fun, but probably a waste of time. I am too obsessed about trading bytes for preciseness anymore and I think it's time to revisit a lot of the implementation code.